### PR TITLE
feat/dotcom: UI tracks Cody Gateway access directly in Enterprise Portal

### PR DESCRIPTION
**Pull request by bobheadxi**
Upgrades the self-contained Cody Gateway modal to only depend on Enterprise Portal to read/write Cody Gateway access.

I also made some minor tweaks to parsing empty form inputs, and also made the "Cody Gateway access token" collapsed by default, with warnings to discourage its use (opted not to remove entirely since it may be useful for dev/debugging)

Depends on:

- https://github.com/sourcegraph/sourcegraph/pull/63926 
- https://github.com/sourcegraph/sourcegraph/pull/63925 
- https://github.com/sourcegraph/sourcegraph/pull/63858 

Closes https://linear.app/sourcegraph/issue/CORE-218

## Test plan

```
sg start dotcom
```

Edit the Cody Gateway access stuff, toggling it up and down. Cross-reference with the `enterprise_portal_cody_gateway_access` table.

### CODE REVIEW

**Comment by bobheadxi** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/64090?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#64057** <a href="https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/64057?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#63959** <a href="https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/63959?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#63927** <a href="https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/63927?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#64087** <a href="https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/64087?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#64090** <a href="https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/64090?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#63926** <a href="https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/63926?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#63925** <a href="https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/63925?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#63858** <a href="https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/63858?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#63837** <a href="https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/63837?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#63812** <a href="https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/63812?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#63831** <a href="https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/63831?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#63737** <a href="https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/63737?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#63864** <a href="https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/63864?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @bobheadxi and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: client/web/src/enterprise/site-admin/dotcom/productSubscriptions/CodyGatewayRateLimitModal.tsx [hunk 1]
```diff
@@ -1,78 +1,103 @@
...
+        setLimit(parseInt(event.target.value || '0', 10))
     }, [])
 
-    const [limitInterval, setLimitInterval] = useState<number>(current?.intervalSeconds ?? 60 * 60)
+    const [limitInterval, setLimitInterval] = useState<Duration>(
+        current?.intervalDuration ?? new Duration({ seconds: BigInt(60 * 60) })
+    )
     const onChangeLimitInterval = useCallback<React.ChangeEventHandler<HTMLInputElement>>(event => {
-        setLimitInterval(parseInt(event.target.value, 10))
+        setLimitInterval(new Duration({ seconds: BigInt(parseInt(event.target.value || '0', 10)) }))
```
Comment (eseliger) [inline client/web/src/enterprise/site-admin/dotcom/productSubscriptions/CodyGatewayRateLimitModal.tsx:37]: I personally would track this as a non-api-layer datatype here, and only make it a Duration when we make the request for storing it, but just preference no need to change

File: client/web/src/enterprise/site-admin/dotcom/productSubscriptions/CodyServicesSection.tsx [hunk 2]
```diff
@@ -92,148 +91,166 @@ export const CodyServicesSection: React.FunctionComponent<Props> = ({
...
+                                    />
+                                </tbody>
+                            </table>
+                            <EmbeddingsRateLimitUsage mode="embeddings" usageQuery={codyGatewayUsageQuery} />
+                        </>
+                    )}
+
+                    <hr className="my-3" />
+                </>
+                <Collapsible titleAtStart={true} title={<H4>Cody Gateway access token</H4>}>
```
Comment (eseliger) [inline client/web/src/enterprise/site-admin/dotcom/productSubscriptions/CodyServicesSection.tsx:221]: is this collapsed by default, or open by default? I think it should be collapsed :)

**Review by eseliger — APPROVED** [review]

File: client/web/src/enterprise/site-admin/dotcom/productSubscriptions/CodyServicesSection.tsx (hunk 2 above)
Comment (bobheadxi) [reply client/web/src/enterprise/site-admin/dotcom/productSubscriptions/CodyServicesSection.tsx:221]: it's collapsed by default from my testing!

File: client/web/src/enterprise/site-admin/dotcom/productSubscriptions/CodyGatewayRateLimitModal.tsx (hunk 1 above)
Comment (bobheadxi) [reply client/web/src/enterprise/site-admin/dotcom/productSubscriptions/CodyGatewayRateLimitModal.tsx]: Fixed! A bit neater that way, thanks

**Pull request merged**